tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js

```diff
@@ -1,25 +1,178 @@
+          )
+          .click('@dateTimePopupYear')
+          .waitForElementVisible(selectors[0])
+          .click(selectors[0])
+          .click('@dateTimeOkButton')
+          .waitForElementVisible('@dateTimePopupDate')
+          .click('@dateTimePopupDate')
+          .waitForElementVisible(selectors[1])
+          .click(selectors[1])
+          .click('@dateTimeOkButton')
+          .waitForElementVisible(selectors[2])
+          .click(selectors[2])
+          .click('@dateTimeOkButton')
+      }
+    },
+    editExpiryDate: async function (linkName) {
+      this.clickLinkEditBtn(linkName)
+      const today = new Date()
+      today.setDate(today.getDate() - 1)
+      const prevDay = today.toDateString().split(/ /)
+      if (prevDay[2][0] === '0') { prevDay[2] = prevDay[2][1] }
+      const selectors = this.setExpiryDateXpath(prevDay[3], prevDay[1], prevDay[2])
+      await this
+        .waitForElementVisible('@linkExpirationDateField')
+        .click('@linkExpirationDateField')
+        .waitForElementVisible('@dateTimePopup')
+        .waitForElementVisible('@dateTimePopupYear')
+        .waitForElementPresent('@dateTimePopupYear')
+
+        .waitForAjaxCallsToStartAndFinish()
+        .waitForElementEnabled(
+          this.elements.dateTimePopupYear.selector
+        )
+        .click('@dateTimePopupYear')
+        .waitForElementVisible(selectors[0])
+        .click(selectors[0])
+        .click('@dateTimeOkButton')
+        .waitForElementVisible('@dateTimePopupDate')
+        .click('@dateTimePopupDate')
+        .waitForElementVisible(selectors[1])
+        .click(selectors[1])
+        .click('@dateTimeOkButton')
+        .waitForElementVisible(selectors[2])
+        .getAttribute(selectors[2], 'class', (result) => {
+          if (result.value.includes('--disabled') === false) {
+            throw new Error('Expected past day is not in disabled state, ' + result)
+          }
+        })
+        .click('@dateTimeOkButton')
+    },
+    editPublicLink: async function (linkName, editData, save = true) {
+      this.clickLinkEditBtn(linkName)
+      for (const [key, value] of Object.entries(editData)) {
+        await this.setPublicLinkForm(key, value)
+      }
+      if (save === true) {
+        await this
+          .waitForElementPresent('@publicLinkSaveButton')
+          .waitForElementVisible('@publicLinkSaveButton')
+          .click('@publicLinkSaveButton')
+          .waitForElementNotPresent('@publicLinkSaveButton')
+          .waitForAjaxCallsToStartAndFinish()
+      }
+    },
+    removePublicLink: async function (linkName) {
+      const linkRowDeleteButtonSelector = util.format(this.elements.publicLinkDeleteButton.selector, linkName)
+      const linkRowDeleteButton = {
+        locateStrategy: 'xpath',
+        selector: linkRowDeleteButtonSelector
+      }
+      await this
+        .waitForElementVisible(linkRowDeleteButton)
+        .click(linkRowDeleteButton)
```

Comment:
Wait for api calls to finish

Reply:
when should we wait?before click or after??